components/calendar/calendar.js

```diff
@@ -696,100 +727,100 @@ class Calendar extends LocalizeCoreElement(RtlMixin(LitElement)) {
+		await this._updateFocusDateOnChange();
 	}
 
-	_onNextMonthButtonClick() {
-		this._updateShownMonthIncrease();
-		this._updateFocusDateOnChange();
+	async _showFocusDateMonth(prevFocusDate, upDownAnimation) {
+		this._isInitialFocusDate = false;
+		const date = await this._getDateElement(this._focusDate);
+		if (!date) {
+			this._keyboardTriggeredMonthChange = true;
+			let increase;
+			if (prevFocusDate < this._focusDate) {
+				increase = true;
+				this._monthIncrease();
+			} else {
+				increase = false;
+				this._monthDecrease();
+			}
+			this._triggerMonthChangeAnimations(increase, true, upDownAnimation);
+		} else {
+			this._focusDateAddFocus();
+		}
+		await this.updateComplete;
+		this._isInitialFocusDate = true;
 	}
 
-	_onPrevMonthButtonClick() {
-		this._updateShownMonthDecrease();
-		this._updateFocusDateOnChange();
+	_triggerMonthChangeAnimations(increase, keyboardTriggered, transitionUpDown) {
+		this._monthNav = undefined;
+		increase = this.dir === 'rtl' ? !increase : increase;
+		if (!keyboardTriggered) this._isInitialFocusDate = true;
+		if (!reduceMotion) {
+			setTimeout(() => {
+				this._monthNav = `${increase ? 'next' : 'prev'}${transitionUpDown ? '-updown' : ''}`;
+			}, 100); // timeout for firefox
+		}
 	}
 
-	async _updateFocusDateDependentOnDisabled(possibleFocusDate, latestPossibleFocusDate) {
+	async _updateFocusDate(possibleFocusDate, latestPossibleFocusDate, allowDisabled) {
 		await this.updateComplete;
-		if (isDateInRange(possibleFocusDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue))) {
+		if (isDateInRange(possibleFocusDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue)) || allowDisabled) {
 			this._focusDate = possibleFocusDate;
+			return true;
 		} else if (this.shadowRoot.querySelector('.d2l-calendar-date:enabled')) {
 			const validDates = this.shadowRoot.querySelectorAll('.d2l-calendar-date:enabled');
 			const focusDate = validDates[latestPossibleFocusDate ? (validDates.length - 1) : 0].parentNode;
 			const year = focusDate.getAttribute('data-year');
 			const month = focusDate.getAttribute('data-month');
 			const date = focusDate.getAttribute('data-date');
 			this._focusDate = new Date(year, month, date);
+			return true;
 		} else {
-			this._focusDate = undefined;
+			return false;
 		}
 	}
 
 	async _updateFocusDateOnChange() {
-		await this.updateComplete;
 		const selectedValueDate = this.selectedValue ? getDateFromISODate(this.selectedValue) : null;
 		const dateElem = selectedValueDate ? await this._getDateElement(selectedValueDate) : null;
-		if (dateElem && isDateInRange(selectedValueDate, getDateFromISODate(this.minValue), getDateFromISODate(this.maxValue))) {
-			this._focusDate = selectedValueDate;
-		} else {
-			await this._updateFocusDateDependentOnDisabled(new Date(this._shownYear, this._shownMonth, 1));
-		}
-	}
-
-	_updateShownMonthDecrease(keyboardTriggered, transitionUpDown) {
```

Comment:
I separated this and `_updateShownMonthIncrease` into separate functions (`_monthIncrease/Decrease` and `_triggerMonthChangeAnimations` so that the arrow key navigation to the next valid date would trigger the animations (even if more than 1 month before/after)